_posts/2024-12-17-release-0_9_0.md

```diff
@@ -0,0 +1,32 @@
+author_url: "https://github.com/SamBarker"
+categories:  [releases, kroxylicious, kafka3.9]
+---
+
+The Kroxylicious project is very pleased to announce the [0.9.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.9.0) release of Kroxylicious. This release introduces support for Apache Kafka 3.9. 
+
+We generally expect Kroxylicious to be both forwards and backwards compatible across Kafka version just like the Apache Kafka client and broker. While we tested with pre-release versions Kafka 3.9 we discovered after it was released that we had missed a case in the protocol version negotiation. When the API Versions request version was updated we were unable to decode the response. In the 0.9.0 release we adopt the same behaviour as the Kafka broker and respond with API version 0 if we do not support the protocol version requested by the client - we have also expanded our test coverage to get earlier warnings of similar issues in the future. 
+
+This release also marks the introduction of our new connection handling sate machine at the core of the proxy. We have evolved state machine at the core of the proxy to separate out the state carried at various stages of the connection life cycle and provide stronger rules around transitions between states. This is still a work in progress as its implications are rather wide-ranging however with think it gives us a stronger core to base everything else around. This does for the moment limit our ability to offer authentication offload support, we do intend to restore this eventually (please do get in touch if this use case is important to you).
+
+There are many quality of life improvements in this release (in no particular order)
+
+- Ensure we now respond with errors instead of closing the connection when there are problems encrypting records 
+- We've made it easier for Filter authors to generate error responses of their own.
+- We realised forwarding partial requests from the record validation filter in the face of validation failures was a bad idea so it now rejects the whole batch.
+- We can now use EC2 instance metadata to authenticate against the AWS KMS. 
+- We now support the validation on M_TLS certificates on the downstream (or client side) of the proxy.
+- Key Management Systems (KMS) now have more life cycle calls backs to better manage shared resources. 
+- We now default to LogJ [Async logging](https://logging.apache.org/log4j/2.x/manual/async.html) - this is mostly a performance optimisation, but we think it should address some issues we have observed with logging in kubernetes environments.
+
+We are very pleased to say we have first time contributions from
+- [Calum Murray](https://github.com/Cali0707)
+- [Alan Robinson](https://github.com/alanrobinson-dwp)
+
+### Feedback
+
+Please let us know, through [Slack](https://kroxylicious.slack.com) or [GitHub](https://github.com/kroxylicious/kroxylicious-junit5-extension/issues), if you find the extension interesting or helpful.
```

Comment:
this is talking about the extension, the prior message for the proxy 0.8.0 was:

`We are eager for any feedback, you can create an [issue in GitHub](https://github.com/kroxylicious/kroxylicious/issues) if you have any problems or want a feature added.` but having the slack link is nice too.

Reply:
Good catch.

Reply:
Agreed, I think the extension wording is nicer but it should reference the proxy instead. How about this?
```suggestion
You can report bugs or request features in Kroxylicious by creating an issue on [GitHub](https://github.com/kroxylicious/kroxylicious-junit5-extension/issues).

Please let us know through [Slack](https://kroxylicious.slack.com) or by tagging the project on social media if you find Kroxylicious interesting or helpful!
```
We'd need to fill in the blanks with links to the Kroxylicious social handles but I think that mostly captures the same energy as the extension feedback text.